### PR TITLE
[coq] Add plugin loader library.

### PR DESCRIPTION
**Pull request by ejgallego**
This is just a copy from SerAPI's one for now, but it belongs here.

### CODE REVIEW

File: coq/loader.ml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    | "coq-core.plugins.ssreflect"        (* ssr *)
+      -> true
+    | _ ->
+      Feedback.msg_warning Pp.(str "Missing serlib plugin: " ++ str fl_pkg);
+      false
+  in
+  if supported
+  then
+    let plugin_name = String.split_on_char '.' fl_pkg |> list_last in
+    Some ("coq-serapi.serlib." ^ plugin_name)
```
Comment (Alizter) [inline coq/loader.ml:41]: Does this need to be renamed?

File: coq/loader.ml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+          Format.eprintf "[plugin loader]: plugin %s requested via findlib@\n%!" fl_pkg;
+        loader [serlib_pkg]
+      | None ->
+        if debug then
+          Format.eprintf "[plugin loader]: plugin %s requested via mltop@\n%!" fl_pkg;
+        loader [fl_pkg]
+    with
+    | Dynlink.Error err as exn ->
+      let msg = Dynlink.error_message err in
+      Format.eprintf "[sertop] Critical Dynlink error %s@\n%!" msg;
```
Comment (Alizter) [inline coq/loader.ml:61]: Also this

File: coq/loader.ml [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+      raise exn
+    | Fl_package_base.No_such_package (pkg, _) as exn ->
+      Format.eprintf "[sertop] Couldn't find the SerAPI plugin %s@; please check `ocamlfind list` does include SerAPI's plugin libraries.\n%!" pkg;
```
Comment (Alizter) [inline coq/loader.ml:64]: and this

File: coq/loader.ml (hunk 1 above)
Comment (ejgallego) [reply coq/loader.ml:41]: No, it does not need renaming. That's a tricky issue as plugins can't register themselves before they are loaded, so for now a manual list needs to be kept.

Comment (ejgallego) [reply coq/loader.ml:41]: Likely we will add a static init to serlib so at least the list is local.

Comment (ejgallego) [reply coq/loader.ml:41]: "No it does not need renaming" really means "no, renaming is wrong and would break everything"

**Pull request merged**